terraformer_policy.tf

```diff
@@ -1,70 +1,32 @@
 # Create the IAM policy for the Terraformer EC2 server instances that
 # allows full access to create/destroy new resources in this account
 # and create/modify/destroy existing resources that _are not_ tagged
-# as being created by the team that deploys this root module.
+# as being created by the team that deploys this root module (there are
+# some exceptions to this rule, see below for details).
 #
 # Also allow sufficient permissions to launch instances in the
 # operations subnet and use the existing security groups.
 
 data "aws_iam_policy_document" "terraformer_policy_doc" {
   provider = aws.provisionassessment
 
-  # Allow full access to new resources and existing resources that
-  # _are not_ tagged as being created by the team that deploys this
-  # root module, with the exception of IAM.
-  #
-  # We will attach the arn:aws:iam::aws:policy/ReadOnlyAccess policy
-  # to the same role to which the policy document will be attached, which
-  # will give it read-only access to all IAM resources.  We require
-  # IAM access to be as read-only as possible in order to stop the
-  # Terraformer instance from defeating the Terraformer policy by
-  # creating new users, policies, roles, etc.
+  # Deny modification of any resources tagged by the team that deploys this
+  # root module, but allow anything else.
   statement {
+    actions = [
+      "*",
+    ]
     condition {
       test = "StringNotEquals"
       values = [
         var.tags["Team"],
       ]
       variable = "aws:ResourceTag/Team"
     }
-    not_actions = [
-      "iam:*",
-    ]
-    resources = [
-      "*",
-    ]
-  }
-
-  # Add an IAM permission to allow the use of our instance roles when
-  # spinning up instances, with the exception of our guacamole, samba,
-  # and terraformer instance roles.  This is one non-read-only IAM
-  # permission that _is_ necessary.
-  statement {
-    actions = [
-      "iam:PassRole",
-    ]
-    condition {
-      test = "StringEquals"
-      values = [
-        "ec2.amazonaws.com",
-      ]
-      variable = "iam:PassedToService"
-    }
     resources = [
       "*",
     ]
   }
-  statement {
-    actions = [
-      "iam:PassRole",
-    ]
-    effect = "Deny"
-    resources = [
-      aws_iam_role.guacamole_instance_role.arn,
-      aws_iam_role.samba_instance_role.arn,
-      aws_iam_role.terraformer_instance_role.arn,
-    ]
-  }
 
   # Allow use of the KMS key used to encrypt COOL AMIs.
```

Comment:
I think we should have the same comments for each of these `statement`s as are used in the corresponding `statement`s in `terraformer_permissionsboundary.tf`.  The latter are more informative, and it's the same code.

Reply:
Good call - updated in https://github.com/cisagov/cool-assessment-terraform/pull/215/commits/df75069c286cd745eac69f96cc66d5cd38772bb0.